create an endpoint for the hello function and a health check endpoint tested the end points with postman

MujtabaNasir opened the pull request:
create an endpoint for the hello function and a health check endpoint
tested the end points with postman

MujtabaNasir committed:
create an endpoint for the hello function and a health check endpoint
tested the end points with postman

mdsrqbl commented on README.md:
````diff
@@ -1 +1,48 @@
+
+4. To deactivate go into the folder where .bat files exists:
+```bash
+cd .\fastapi_env\Scripts\
````
use forward slashes

MujtabaNasir replied:
Done

mdsrqbl commented on README.md:
````diff
@@ -1 +1,48 @@
+4. To deactivate go into the folder where .bat files exists:
+```bash
+cd .\fastapi_env\Scripts\
+run command: deactivate
````
avoid writing instructions inside code blocks

MujtabaNasir replied:
Done

mdsrqbl commented on README.md:
````diff
@@ -1 +1,48 @@
+
+5. Install necessary dependancies
+```bash
+pip install fastapi uvicorn
````
use `-r requirements.txt` istead

MujtabaNasir replied:
Done

mdsrqbl commented on README.md:
````diff
@@ -1 +1,48 @@
+
+6. Install mujtaba-charm package:
+```bash
+pip install git+https://github.com/MujtabaNasir/mujtaba-charm.git
````
add to your requirements file

MujtabaNasir replied:
Done

mdsrqbl commented:
integrate python logging into your service

MujtabaNasir committed:
GitHub reviews resolved
Python logging implemented

mdsrqbl commented on main.py:
```diff
@@ -0,0 +1,49 @@
+    logger.info("Handling /hello request")
+    message = hello()
+    logger.info(f"Returning message: {message}")
+    return {"message": message}
```
this is code duplication

mdsrqbl commented on main.py:
```diff
@@ -0,0 +1,49 @@
+    return {"message": message}
+
+@app.get("/hello/{name}")
+async def get_hello(name):
```
make the parameter optional to handle thee senario where parameter isn't passed

mdsrqbl commented on main.py:
```diff
@@ -0,0 +1,49 @@
+    return {"status": "ok"}
+
+if __name__=="__main__":
+    uvicorn.run(app,host="0.0.0.0",port=8000)
```
format the code properly
(use black)

mdsrqbl commented on main.py:
```diff
@@ -0,0 +1,49 @@
+from fastapi import FastAPI
+import uvicorn
+from mujtaba_charm.utils.sample import hello
+import logging 
```
sort your imports properly

MujtabaNasir committed:
Code duplication removed
black and isort done
git+ tested to work fine by pip install -r requirements.txt

mdsrqbl commented on main.py:
```diff
@@ -0,0 +1,50 @@
+logger = logging.getLogger(__name__)
+
+
+@app.get("/hello")
```
do you really need to write this endpoint twice?

MujtabaNasir replied:
@app.get("/hello") outputs "Hello World!"
with the same endpoint it outputs "Hello {name}!" with the browser url of http://127.0.0.1:8000/hello?name=Jack

mdsrqbl commented on main.py:
```diff
@@ -0,0 +1,50 @@
+    Returns:
+        dict: A dictionary containing a greeting message.
+    """
+    if name:
+        logger.info(f"Handling /hello/{name} request")
+        message = hello(name)
+    else:
+        logger.info("Handling /hello request")
+        message = hello()
```
remove code duplication

explore python's `or` operator to avoid the if/else

```python
hello(name or "my-dear-user")
```

MujtabaNasir replied:
Done

MujtabaNasir committed:
Code duplication removed.
Job done with just one endpoint now.
